src/pipecat/processors/aggregators/dtmf_aggregator.py

```diff
@@ -0,0 +1,162 @@
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection) -> None:
+        await super().process_frame(frame, direction)
+
+        if isinstance(frame, (EndFrame, CancelFrame)):
+            # Flush any pending aggregation before stopping
+            if self._aggregation:
+                await self._flush_aggregation()
+            await self._stop()
+            await self.push_frame(frame, direction)
+            return
+
+        # Push all other frames downstream immediately
+        await self.push_frame(frame, direction)
+
+        if isinstance(frame, InputDTMFFrame):
+            await self._handle_dtmf_frame(frame)
+
+    async def _handle_dtmf_frame(
+        self,
+        frame: InputDTMFFrame,
+    ):
+        # Create task on first DTMF input if needed
+        if not self._aggregation_task:
+            self._create_aggregation_task()
+
+        digit_value = frame.button.value
+        self._aggregation += digit_value
+
+        # If this is the first digit, send BotInterruptionFrame upstream
+        # But use a separate task to avoid interfering with current frame processing
+        if len(self._aggregation) == 1:
+            # Use create_task to avoid queue issues
+            self.create_task(self._send_interruption_frame())
+
+        # Check for immediate flush conditions
+        if frame.button == self._termination_digit:
+            await self._flush_aggregation()
+        else:
+            # Signal new digit received
+            self._digit_event.set()
+
+    async def _send_interruption_frame(self):
+        """Send an interruption frame in a separate task to avoid queue issues.
+
+        This interruption frame allows for the user to interrupt the bot's speaking
+        with a keypress. Without this, the TranscriptionFrame generated is accompanied
+        by EmulatedUserStarted/StoppedSpeakingFrames, which the bot currently ignores.
+        This treats the keypress as an explicit input which the bot should respond to.
+        """
+        try:
+            await self.push_frame(BotInterruptionFrame(), FrameDirection.UPSTREAM)
+        except Exception as e:
+            print(f"Error sending interruption frame: {e}")
+
+    async def _aggregation_task_handler(self):
+        """Background task that handles timeout-based flushing."""
+        try:
+            while True:
+                try:
+                    await asyncio.wait_for(self._digit_event.wait(), timeout=self._idle_timeout)
+                except asyncio.TimeoutError:
+                    if self._aggregation:
+                        await self._flush_aggregation()
+                finally:
+                    self._digit_event.clear()
+        except asyncio.CancelledError:
+            # Task was cancelled - exit cleanly
+            pass
+        except Exception as e:
+            # Log unexpected errors but don't crash
+            print(f"Unexpected error in DTMF aggregation task: {e}")
```

Comment:
you just need the while loop. we internally capture `CancelledError`. also, when you capture `CancelledError` you need to re-raised it. But anyways, I think just the while loop would be fine and shorter.